File: ruby/intro/hello.js
// This is a comment.

console.log(4 * 4); // This would evaluate to 8.

function square(n) {
  if (n > 7) console.log("It's huge!");
  return n * n;
}

function double(n) {
  if (!(n % 2 === 0)) console.log(`${n} is odd!`);
  return n * 2;
}

// const square = n => n * n;

console.log(square(7));
console.log(square(8));

console.log(double(5));
console.log(double(6));

let n = 0;

// n++;

n += 1;

console.log(n);

n = 42;

console.log(n);

const PHI = 1.61803399;

// PHI = 1.618;

console.log(3 * PHI);

const fibb = [1, 1, 2, 3, 5, 8, 13, 21, 34];

for (let i = 0; i < fibb.length; i++) {
  console.log(fibb[i]);
}

fibb.forEach(n => console.log(n));

const eyeColors = {
  "Matt C": "Brown",
  "Karsten R": "Blue",
  "Ben J": "Grey",
  "Jason P": "Hazel"
};

console.log(eyeColors);

File: ruby/intro/hello.rb
# This is a comment.

puts(4 * 4) # This would evaluate to 8.

def square(n)
  puts "It's huge!" if n > 7
  n * n
end

puts square(7)
puts square(8)

def double(n)
  puts "#{n} is odd." unless n.even?
  n * -18
end

puts double(15)
puts double(6)

n = 0
# n += 1

puts n

n = 42

puts n

PHI = 1.61803399 # This is a constant

puts 3 * PHI

FIBB = [1, 1, 2, 3, 5, 8, 13, 21, 34].freeze

# Uncommon practice.
for n in FIBB
  puts n
end

n = ""

while n != "nnnn"
  puts "#{n} is not nnnn!"
  n += "n"
end

FIBB.each do |n|
  puts n
end

FIBB.each { |n| puts n }

FIBB.map { |n| double(n) }.each { |n| puts "#{n} is the double of #{n / 2}" }

eye_colors = {
  matt: "Brown",
  karsten: "Blue",
  ben: "Grey",
  jason: "Hazel",
}

puts eye_colors

foo = "foo"

puts foo
puts :foo

puts "I am going to double the number you give me. I can math!"

number = gets.chomp

puts "You gave me: \"#{number}\". Now I will double it."

puts double(number.to_i)

puts square(72_000) # Instead of commas, we can use underscores.
